fetch-in-react/my-app/src/User.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import UserCard from './UserCard';
+
+export default function User({ userId, onCancel }) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+  const [user, setUser] = useState();
+
+  /* your code here (hint: useEffect) */
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await fetch(`https://jsonplaceholder.typicode.com/users/${userId}`);
+        if (!response.ok) {
+          throw new Error(`Network response was not OK! Error status:${response.status}`);
+        }
+        const jsonData = await response.json();
+        setUser(jsonData);
+        setIsLoading(false);
+      } catch (err) {
+        setError(err.message);
+        setIsLoading(false);
+      }
+    })();
+    return () => {
```

Comment:
Don't do this. Resetting your state variables in an Effect clean-up is not a good practice as it can lead to bugs when the state resets unexpectedly.